File: 3838-weighted-word-mapping/3838-weighted-word-mapping.py
class Solution:
    def mapWordWeights(self, words: List[str], weights: List[int]) -> str:
        ans = []

        for word in words:
            curr_total = 0

            for c in word:
                curr_total += weights[(ord(c) - ord('a'))]

            curr_total = 25 - (curr_total % 26)
            ans.append(chr(ord('a') + curr_total))

        return "".join(ans)

File: 3838-weighted-word-mapping/3838-weighted-word-mapping.java
class Solution {
    public String mapWordWeights(String[] words, int[] weights) {
        StringBuilder ans = new StringBuilder(words.length);

        for (String word : words){
            int curr_total = 0;

            for (int i = 0; i < word.length(); i++){
                curr_total += weights[word.charAt(i) - 'a'];
            }

            ans.append((char) (25 - (curr_total % 26) + 'a'));
        }

        return ans.toString();
    }
}